tests/sync/app.cpp

```diff
@@ -1908,6 +1908,67 @@ TEST_CASE("app: sync integration", "[sync][app]") {
         util::EventLoop::main().run_until([&]{ return error_did_occur.load(); });
         REQUIRE(error_did_occur.load());
     }
+
+    SECTION("invalid pk schema error handling")
+    {
+        TestSyncManager sync_manager({.app_config = app_config});
+        auto app = get_app_and_login(sync_manager.app());
+
+        const auto dog_schema = realm::ObjectSchema("Dog", {
+            realm::Property("my_primary_key", PropertyType::ObjectId | PropertyType::Nullable, true),
+            realm::Property("breed", PropertyType::String | PropertyType::Nullable),
+            realm::Property("name", PropertyType::String),
+            realm::Property("realm_id", PropertyType::String | PropertyType::Nullable)
+        });
+        const auto person_schema = realm::ObjectSchema("Person", {
+            realm::Property("_id", PropertyType::ObjectId | PropertyType::Nullable, true),
+            realm::Property("age", PropertyType::Int),
+            realm::Property("dogs", PropertyType::Object | PropertyType::Array, "Dog"),
+            realm::Property("firstName", PropertyType::String),
+            realm::Property("lastName", PropertyType::String),
+            realm::Property("realm_id", PropertyType::String | PropertyType::Nullable)
+        });
+        Schema schema = {dog_schema, person_schema};
```

Comment:
This should define the schema in the same way as all of the other tests rather than doing something different.

Reply:
Updated to use the existing schema and modify the properties manually which should make it more clear what is being tested instead of having to read the schema diff.